### PR TITLE
moved variables that could be defined by inventory out of common

### PR DESCRIPTION
**Pull request by jdextercsc**

### CODE REVIEW

File: inventory/group_vars/all/common.yml [hunk 1]
```diff
@@ -99,18 +82,14 @@ controller_nodes:
 
 
 ### Messaging
-rabbit_host: 172.17.17.99
-rabbit_hosts: 172.17.17.70:5672,172.17.17.71:5672,172.17.17.72:5672
-rabbit_port: 5672
```
Comment (jcpowermac) [inline inventory/group_vars/all/common.yml:104]: I think `rabbit_port` should be in defaults my only concern is when for example we just want a compute node built if the default variable will be added.  I guess we just need to test more.

File: roles/ceilometer-common/vars/main.yml [hunk 2]
```diff
@@ -2,3 +2,4 @@
 public_ipaddr: "{{ hostvars[inventory_hostname][ 'ansible_' + primary_if ].ipv4.address }}"
 internal_ipaddr: "{{ hostvars[inventory_hostname][ 'ansible_' + private_if ].ipv4.address }}"
 ceilometer_bind_host: "{{ internal_ipaddr }}"
+mongodb_servers: "{% for node in groups['controller'] %}{{ hostvars[node]['inventory_hostname'] }}:{{ mongodb_port | default(27017) }}{% if not loop.last %},{% endif %}{% endfor %}"
```
Comment (jcpowermac) [inline roles/ceilometer-common/vars/main.yml]: Are we after hostname here or IP address?

If hostname should be using `ansible_hostname`
If IP address it should be `['ansible_'+private_if]['ipv4']['address']`

File: roles/galera/defaults/main.yml [hunk 3]
```diff
@@ -2,3 +2,5 @@
 use_galera: true
 use_netapp_copyoffload: false
 clustercheck_db_user: clustercheck
+wsrep_cluster_name: galera_cluster
+wsrep_cluster_address: "gcomm://{{ {% for node in groups['controller'] %}{{ hostvars[node]['inventory_hostname'] }}{% if not loop.last %},{% endif %}{% endfor %} }}"
```
Comment (jcpowermac) [inline roles/galera/defaults/main.yml]: Same as above

File: roles/mongodb/vars/main.yml [hunk 4]
```diff
@@ -2,3 +2,4 @@
 public_ipaddr: "{{ hostvars[inventory_hostname][ 'ansible_' + primary_if ].ipv4.address }}"
 internal_ipaddr: "{{ hostvars[inventory_hostname][ 'ansible_' + private_if ].ipv4.address }}"
 mongodb_bind_host: "{{ internal_ipaddr }}"
+mongo_replSet: "{% for node in groups['controller'] %}{% if hostvars[node]['inventory_hostname'] not hostvars[groups['controller'][0]]['inventory_hostname'] %} {{hostvars[node]['inventory_hostname']}}{% endif %}{% endfor %}"
```
Comment (jcpowermac) [inline roles/mongodb/vars/main.yml]: Same as above.  The problem being that `inventory_hostname` is not in the dictionary.

See example dictionary items from the following command.
`ansible -m setup 172.17.16.70 -u root`

File: inventory/group_vars/all/common.yml (hunk 1 above)
Comment (jdextercsc) [inline inventory/group_vars/all/common.yml:104]: rabbit_port was removed from there.  I can add in {{ rabbit_port | default (5672) }}

Comment (jcpowermac) [inline inventory/group_vars/all/common.yml:104]: That would work, maybe that should be a "best practice" of ours.

**Comment by jcpowermac** [comment]
Code review :+1:

**Comment by adean20csc** [comment]
Good  :+1:

**Pull request merged**